packages/buildcop/buildcop.go

```diff
@@ -0,0 +1,208 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Command buildcop searches for sponge_log.xml files and publishes them to
+// Pub/Sub.
+//
+// You can run it locally by running:
+//   go build
+//   ./buildcop -repo=my-org/my-repo -installation_id=123 -project=my-project
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+	"google.golang.org/api/option"
+)
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix("[Buildcop] ")
+	log.SetOutput(os.Stderr)
+
+	repo := flag.String("repo", "", "The repo this is for. Defaults to auto-detect from Kokoro environment. If that doesn't work, if your repo is github.com/GoogleCloudPlatform/golang-samples, --repo should be GoogleCloudPlatform/golang-samples")
+	installationID := flag.String("installation_id", "", "GitHub installation ID. Defaults to auto-detect. If your repo is not part of GoogleCloudPlatform or googleapis set this to the GitHub installation ID for your repo. See https://github.com/googleapis/repo-automation-bots/issues.")
+	projectID := flag.String("project", "repo-automation-bots", "Project ID to publish to. Defaults to repo-automation-bots.")
+	topicID := flag.String("topic", "passthrough", "Pub/Sub topic to publish to. Defaults to passthrough.")
+
+	flag.Parse()
+
+	log.Println("Sending logs to Build Cop Bot...")
+	if ok := publish(*projectID, *topicID, *repo, *installationID); !ok {
+		os.Exit(1)
+	}
+	log.Println("Done!")
+}
+
+type githubInstallation struct {
+	ID string `json:"id"`
+}
+
+type message struct {
+	Name         string
+	Type         string
+	Location     string
+	installation githubInstallation
+	repo         string
+	buildID      string
+	buildURL     string
+	xunitXML     string
+}
+
+// publish searches for sponge_log.xml files and publishes them to Pub/Sub.
+// publish logs a message and returns false if there was an error.
+func publish(projectID, topicID, repo, installationID string) (ok bool) {
+	ctx := context.Background()
+
+	gfileDir := os.Getenv("KOKORO_GFILE_DIR")
+	if gfileDir == "" {
+		log.Println("KOKORO_GFILE_DIR not set, unable to get service account")
+		return false
+	}
+	saPath := filepath.Join(gfileDir, "kokoro-trampoline.service-account.json")
+
+	client, err := pubsub.NewClient(ctx, projectID, option.WithCredentialsFile(saPath))
+	if err != nil {
+		log.Printf("Unable to connect to Pub/Sub: %v", err)
+		return false
+	}
+	topic := client.Topic(topicID)
+
+	if repo == "" {
+		var ok bool
```

Comment:
nit: consider different name or don't use named return to avoid shadow and having to redeclare var name

Reply:
Removed the pre-declaration. Want the named return for documentation purposes.